src/lib/response-triggers/location.ts

```diff
@@ -0,0 +1,111 @@
+import Trigger from '../../models/trigger';
+import { DbLocation } from '../database/db-interfaces';
+import Config from '../config-interface';
+import { MessageObject } from 'webex/env';
+import { Role } from '../../models/database';
+import { Command } from '../../models/command';
+import constants from '../../constants';
+import xmlMessageParser from '../parsers/xmlMessageParser';
+import { LocationAction } from '../../models/location-action';
+import { Logger } from '../logger';
+
+export default class Location extends Trigger {
+	dbLocation : DbLocation;
+	config : Config;
+
+	constructor(dbLocation : DbLocation, config : Config) {
+		super();
+
+		this.dbLocation = dbLocation;
+		this.config = config;
+	}
+
+	isToTriggerOn(message : MessageObject) : boolean {
+		let parsedMessage = xmlMessageParser.parseNonPegMessage(message);
+		return parsedMessage.botId === constants.botId && parsedMessage.command.toLowerCase().startsWith(Command.Location);
+	}
+
+	async createMessage(message : MessageObject) : Promise<MessageObject> {
+		let parsedMessage = xmlMessageParser.parseNonPegMessage(message);
+		let words = parsedMessage.command.trim().split(' ').filter(x => x !== '');
+
+		let response : string;
+
+		if (words.length < 2) {
+			return { markdown: `Please specify a command. Possible values are ${Object.values(LocationAction).join(', ')}` };
+		}
+
+		let locations: string[];
+		try {
+			locations = await this.dbLocation.getLocations();
+		} catch (error) {
+			Logger.error(`[Location.createMessage] Error getting locations ${error.message}`);
+			return { markdown: 'Error getting locations' };
+		}
+
+		switch (words[1].toLowerCase()) {
+			case LocationAction.Get:
+				response = await this.getLocationMessage(locations);
+				break;
+			case LocationAction.Set:
+				if (!(this.config.checkRole(message.personId, Role.Admin) || this.config.checkRole(message.personId, Role.Location))) {
```

Comment:
Could we break each of these cases up into separate functions for clarity?

Reply:
Done